examples/crowdfunding/test/crowdfundingTest.js

```diff
@@ -0,0 +1,272 @@
+import {
+  addressToBytes,
+  getProgram,
+  intToBigEndian,
+  SignType,
+  stringToBytes,
+  TransactionType
+} from '@algorand-builder/algob';
+import { Runtime, StoreAccountImpl } from '@algorand-builder/algorand-js/build/src/index';
+import { assert } from 'chai';
+
+import { getAcc } from './common';
+
+const initialDonorBalance = 60000000;
+const initialCreatorBalance = 10000;
+const goal = 7000000;
+
+describe('Crowdfunding Tests', function () {
+  let creator = new StoreAccountImpl(initialCreatorBalance);
+  let escrow = new StoreAccountImpl(0);
+  let donor = new StoreAccountImpl(initialDonorBalance);
+
+  let runtime;
+  let flags;
+  const program = getProgram('crowdFundApproval.teal');
+
+  this.beforeAll(async function () {
+    runtime = new Runtime([creator, escrow, donor]);
+
+    flags = {
+      sender: creator.account,
+      localInts: 1,
+      localBytes: 0,
+      globalInts: 5,
+      globalBytes: 3
+    };
+  });
+
+  this.afterEach(async function () {
+    creator = new StoreAccountImpl(initialCreatorBalance);
+    escrow = new StoreAccountImpl(0);
+    donor = new StoreAccountImpl(initialDonorBalance);
+    runtime = new Runtime([creator, escrow, donor]);
+
+    flags = {
+      sender: creator.account,
+      localInts: 1,
+      localBytes: 0,
+      globalInts: 5,
+      globalBytes: 3
+    };
+  });
+
+  // update account state
+  function syncAccounts () {
+    creator = getAcc(runtime, creator);
+    escrow = getAcc(runtime, escrow);
+    donor = getAcc(runtime, donor);
+  }
+
+  // Get begin date to pass in
+  const beginDate = new Date();
+  beginDate.setSeconds(beginDate.getSeconds() + 2);
+
+  // Get end date to pass in
+  const endDate = new Date();
+  endDate.setSeconds(endDate.getSeconds() + 12000);
+
+  // Get fund close date to pass in
+  const fundCloseDate = new Date();
+  fundCloseDate.setSeconds(fundCloseDate.getSeconds() + 120000);
+
+  const creationArgs = [
+    intToBigEndian(beginDate.getTime()),
+    intToBigEndian(endDate.getTime()),
+    intToBigEndian(goal),
+    addressToBytes(creator.address),
+    intToBigEndian(fundCloseDate.getTime())
+  ];
+
+  it('crowdfunding application', async () => {
```

Comment:
This is a long test. Please describe the test scenario just below the `it(...` statement in a comment.

Reply:
ok